File: JavaScript/05. Conditional-Statements/ConditionalStatements-Homework/ConditionalStatements-Homework/06. QuadraticEquation/QuadraticEquation.js
/*
 * Problem 6. Quadratic equation

Write a script that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and solves it (prints its real roots).
Calculates and prints its real roots.
Note: Quadratic equations may have 0, 1 or
 */

function quadraticEquation() {
    var a = parseFloat(document.getElementById('a').value);
    var b = parseFloat(document.getElementById('b').value);
    var c = parseFloat(document.getElementById('c').value);
    var x1 = 0, x2 =0 , D =0;

    console.log(b);

    if (!isNaN(a) && !isNaN(b) && !isNaN(c)) {
        D = Math.pow(b, 2) - (4 * a * c);
        if (D === 0) { // one real root
            x1 = (-b) / (2 * a);
            console.log("x1 = x2 = " + x1);
        } else if(D > 0) { // two real roots
            x1 = ((-b) - Math.sqrt(D)) / (2 * a);
            x2 = ((-b) + Math.sqrt(D)) / (2 * a);
            console.log("x1 = " + x1 + ",  x2 = " + x2);
        } else {
            console.log("no real roots");
        }
    } else {
        console.log("a or b or c is not a number!");
    }
}

File: C# I/04. Console-Input-Output/Console-Input-Output-Homework/06. QuadraticEquation/QuadraticEquation.cs
using System;

class QuadraticEquation
{
    /* 6. Quadratic Equation

    Write a program that reads the coefficients a, b and c of a quadratic 
    equation ax2 + bx + c = 0 and solves it (prints its real roots).
    */

    static void Main()
    {
        Console.Title = "Quadratic Quation";
        Console.Write("a = ");
        double a = double.Parse(Console.ReadLine());
        Console.Write("b = ");
        double b = double.Parse(Console.ReadLine());
        Console.Write("c = ");
        double c = double.Parse(Console.ReadLine());
        Console.WriteLine("The quadratic equation is:   {0}x^2 + {1}x + {2} = 0", a, b, c);

        double discriminant = (b * b) - (4 * (a * c));
        if(discriminant < 0)
        {
            Console.WriteLine("no real roots");
        }
        else if(discriminant == 0)
        {
            double oneRoot = (b * (-1)) / (2 * a);
            Console.WriteLine("The eqiation has only one root. x1 = x2 = " + oneRoot);
        }
        else
        {
            double x1 = ((b * (-1)) - Math.Sqrt(discriminant)) / (2 * a);
            double x2 = ((b * (-1)) + Math.Sqrt(discriminant)) / (2 * a);
            Console.WriteLine("The roots are:   x1 = {0} x2 = {1}",x1,x2);
        }
    }
}

